sgt-back-end/index.js

```diff
@@ -0,0 +1,128 @@
+const express = require('express');
+const pg = require('pg');
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+app.use(express.json());
+
+app.get('/api/grades', (req, res) => {
+  const sql = `
+    select *
+    from grades;
+  `;
+
+  db.query(sql)
+    .then(result => {
+      const grades = result.rows;
+      res.status(200).json(grades);
+    })
+    // eslint-disable-next-line node/handle-callback-err
+    .catch(error => {
+      res.status(500).json({ error: 'An unexpected error occurred' });
+    });
+
+});
+
+app.post('/api/grades', (req, res) => {
+  const { name, course, score } = req.body;
+
+  const sql = `
+    insert into "grades" ("nme","course","score")
+    values ('${name}','${course}','${score}')
+    returning *;
+  `;
+  if (!name || !course || !score || score < 0 || score > 100) {
+    res.status(400).json({ error: 'Name, Course, and Score are required' });
+  }
+
+  db.query(sql).then(result => {
+    res.status(200).json({ name, course, score });
+  // eslint-disable-next-line node/handle-callback-err
+  }).catch(error => {
+    res.status(500).json({ error: 'An unexpected error occurred' });
+  });
+});
+
+app.put('/api/grades/:gradeId', (req, res) => {
+  const { name, course, score } = req.body;
+  const id = +req.params.gradeId;
+
+  const sqlChecking = `
+  select "name","course","score"
+  from "grades"
+  where "gradeId" = ${id};
+  `;
+
+  const sql = `
+  update "grade"
+  set "name"='${name}',"course"='${course}',"score"='${score}'
+  where "gradeId"=${id}
+  returning *;
+  `;
+
+  if (!name || !course || !score || score < 0 || score > 100) {
+    res.status(400).json({ error: 'Name, Course, and Score are required' });
+  } else {
+    db.query(sqlChecking)
```

Comment:
You don't need to do a checking query first, if the `gradeId` isn't a valid `gradeId` the update query will fail and not send back any data from the update query. You do want to validate that the `gradeId` is an integer that is greater than zero before making the update query